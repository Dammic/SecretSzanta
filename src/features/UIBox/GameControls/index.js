export { default as GameControls } from './GameControls'
