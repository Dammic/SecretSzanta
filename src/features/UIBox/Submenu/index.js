export { default as Submenu } from './Submenu'
