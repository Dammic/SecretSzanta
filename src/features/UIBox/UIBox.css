@import 'variables/variables.css';

.uiBox {
    position: relative;
    bottom: 0;
    width: 0;
    z-index: 30;
    height: 20%;
    max-height: 200px;
}

.gameControls {
    display: flex;
    width: 75px;
    height: 100%;
    z-index: 20;
    background-color: var(--uibox-background);
    position: absolute;
}

.buttons {
    display: flex;
    flex-direction: column;
    width: 50px;
}
.sideButtons {
    display: flex;
    flex-direction: column;
}

.submenu {
    background-color: var(--uibox-underbackground);
    right: 0;
    top: 0;
    position: absolute;
    width: 218px;
    height: 100%;
    left: 75px;
    transition: all 0.5s;
    &.hidden {
        left: -300px;
    }
}

.affiliationCards {
    display: flex;
    height: 80%;
    & .card {
        flex-basis: 0;
        flex-grow: 1;
        background-repeat: no-repeat;
        background-position: center;
        background-size: 100% 100%;
        &:first-child {
            margin-right: 5px; 
        }
        &.liberal {
            background-image: url('../../static/Liberal_Membership.png');
        }

        &.facist {
            background-image: url('../../static/Facist_Membership.png');
        }
        & .uiBoxAvatar {
            height: 100%;
            & .portrait {
                max-width: unset;
                width: 100%;
                height: inherit;
            }
        }
    }
}
.roleWrapper {
    width: 100%;
    display: flex;
    justify-content: center;
    height: 20%;
    align-items: center;
    & .uiBoxRole {
        width: 100%; 
    }
}
