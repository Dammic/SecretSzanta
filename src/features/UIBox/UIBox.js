import React from 'react'
import UIBoxComponent from './UIBoxComponent'
import { SocketEvents } from '../../../Dictionary'
import { socket } from '../../utils/SocketHandler'

export default class UIBox extends React.PureComponent {
    onStartVote = () => {
        socket.emit(SocketEvents.CHANCELLOR_CHOICE_PHASE)
    }

    onStartGame = () => {
        socket.emit(SocketEvents.START_GAME)
    }

    render () {
        return (
            <UIBoxComponent onStartVote={this.onStartVote} onStartGame={this.onStartGame}/>
        )
    }
}
