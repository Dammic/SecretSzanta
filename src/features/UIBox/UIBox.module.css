@import 'variables/variables.css';

.uiBox {
    position: relative;
    bottom: 0;
    width: 0;
    z-index: 30;
    height: 20%;
    max-height: 200px;
}
