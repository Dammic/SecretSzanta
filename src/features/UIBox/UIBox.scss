.ui-box {
    position: relative;
    bottom: 0;
    width: 0;
    z-index: 30;
    height: 20%;
    max-height: 200px;
    background-color: $uibox-background;

    .game-controls {
        display: flex;
        width: 75px;
        height: 100%;
        z-index: 20;
        background-color: #E1F2F1;
        position: absolute;
        .buttons {
            display: flex;
            flex-direction: column;
            width: 50px;
        }
        .side-buttons {
            display: flex;
            flex-direction: column;
        }
    }

    .affiliation-cards {
        background-color: yellow;
        display: inline-block;
        right: 0;
        top: 0;
        position: absolute;
        width: 218px;
        height: 100%;
        left: 75px;
        transition: all 0.5s;
        &.hidden {
            left: -300px;  
        }
    }
}
