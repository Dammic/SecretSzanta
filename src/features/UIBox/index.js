export { default as UIBox } from './UIBox'
