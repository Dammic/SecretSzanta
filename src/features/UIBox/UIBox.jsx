import React from 'react'
import GameControls from './GameControls/GameControls'
import Submenu from './Submenu/Submenu'

import styles from './UIBox.css'

const UIBox = () => (
    <div className={styles.uiBox}>
        <GameControls />
        <Submenu />
    </div>
)

UIBox.displayName = 'UIBox'
export default UIBox
