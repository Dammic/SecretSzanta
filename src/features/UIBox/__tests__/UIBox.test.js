import React from 'react'
import { expectShallowMatchingSnapshot } from 'packages/testUtils'
import UIBox from '../UIBox'

describe('<UIBox />', () => {
    it('matches snapshot', () => {
        expectShallowMatchingSnapshot(<UIBox />)
    })
})
