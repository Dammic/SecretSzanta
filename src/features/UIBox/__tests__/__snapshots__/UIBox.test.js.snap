// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UIBox /> matches snapshot 1`] = `
<div
  className="uiBox"
>
  <Connect(GameControls) />
  <Connect(Submenu) />
</div>
`;
